src/libs/actions/IOU.ts

```diff
@@ -6842,7 +6842,14 @@ function approveMoneyRequest(expenseReport: OnyxEntry<OnyxTypes.Report>, full?:
         total = expenseReport?.unheldTotal;
     }
     const optimisticApprovedReportAction = ReportUtils.buildOptimisticApprovedReportAction(total, expenseReport?.currency ?? '', expenseReport?.reportID ?? '-1');
-    const optimisticNextStep = NextStepUtils.buildNextStep(expenseReport, CONST.REPORT.STATUS_NUM.APPROVED);
+
+    const approvalChain = ReportUtils.getApprovalChain(PolicyUtils.getPolicy(expenseReport?.policyID), expenseReport?.ownerAccountID ?? -1, expenseReport?.total ?? 0);
+    const isLastApprover = approvalChain[approvalChain.length - 1] === currentUserEmail ?? true;
```

Comment:
Hmm I don't think `?? true` does anything here... `??` only checks if the left side is `undefined` or `null`, right?? But in our case the left side of `??` is always a boolean. `false ?? true` evaluates to `false`.

Reply:
yes that is the intent. if left side is null / undefined then set it to `true`

Reply:
I'll admit that this line of code is not easy to read. Let me refactor to better express the intent

Reply:
@Beamanator the code is equivalent to - 

```js
function isLastApprover(approvalChain: string[]): boolean {
    if (approvalChain.length === 0) {
        return true;
    }
    return approvalChain[approvalChain.length - 1] === currentUserEmail;
}

```